community/kernel-api/src/test/java/org/neo4j/internal/kernel/api/StubPropertyCursor.java

```
@@ -0,0 +1,179 @@
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.internal.kernel.api;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import org.neo4j.values.storable.Value;
+import org.neo4j.values.storable.ValueGroup;
+import org.neo4j.values.storable.ValueWriter;
+
+public class StubPropertyCursor implements PropertyCursor
+{
+    private int offset = -1;
+    private Integer[] keys;
+    private Value[] values;
+
+    void init( Map<Integer, Value> properties )
+    {
+        offset = -1;
+        keys = properties.keySet().toArray( new Integer[0] );
+        values = properties.values().toArray( new Value[0] );
+    }
+
+    @Override
+    public boolean next()
+    {
+        return ++offset < keys.length;
+    }
+
+    @Override
+    public boolean shouldRetry()
+    {
+        return false;
+    }
+
+    @Override
+    public void close()
+    {
+
+    }
+
+    @Override
+    public int propertyKey()
+    {
+        return keys[offset];
+    }
+
+    @Override
+    public ValueGroup propertyType()
+    {
+        return values[offset].valueGroup();
+    }
+
+    @Override
+    public Value propertyValue()
+    {
+        return values[offset];
+    }
+
+    @Override
+    public <E extends Exception> void writeTo( ValueWriter<E> target )
+    {
+
```

We should throw `UnsupportedOperationException` here, otherwise it will bite us later.